### PR TITLE
[gtest] conversion for code coverage tests

### PR DESCRIPTION
**Pull request by cderb**

### CODE REVIEW

**Review by DrizztDoUrden — APPROVED** [review]

**Review by junliume — APPROVED** [review]

**Comment by junliume** [comment]
@cderb it seems that after this change "Fp32 Hip Debug gfx908" stage gets stuck every time ... investigating.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -776,14 +762,6 @@ set(TEST_CONV_VERBOSE_F ${MIOPEN_TEST_FLOAT_ARG} --verbose --disable-backward-da
 set(TEST_CONV_VERBOSE_B ${MIOPEN_TEST_FLOAT_ARG} --verbose --disable-forward --disable-backward-weights)
 set(TEST_CONV_VERBOSE_W ${MIOPEN_TEST_FLOAT_ARG} --verbose --disable-forward --disable-backward-data)
 
-add_custom_test(test_pooling2d_asymmetric SKIP_UNLESS_ALL HALF_ENABLED GFX94X_ENABLED GFX103X_ENABLED GFX110X_ENABLED
-    COMMAND $<TARGET_FILE:test_pooling2d> ${MIOPEN_TEST_FLOAT_ARG} --all --dataset 1 --limit 0 ${MIOPEN_TEST_FLAGS_ARGS}
-)
-
-add_custom_test(test_pooling2d_wide SKIP_UNLESS_ALL HALF_ENABLED GFX94X_ENABLED GFX103X_ENABLED GFX110X_ENABLED
-    COMMAND $<TARGET_FILE:test_pooling2d> ${MIOPEN_TEST_FLOAT_ARG} --all --dataset 2 --limit 0 ${MIOPEN_TEST_FLAGS_ARGS}
-)
```
Comment (atamazov) [inline test/CMakeLists.txt:785]: [Recommendation] This is not a code coverage test. The title of PR should be changed.

File: test/driver.hpp [hunk 2]
```diff
@@ -1381,7 +1386,7 @@ template <template <class...> class Driver>
 void test_drive(int argc, const char* argv[])
 {
     std::vector<std::string> as(argv + 1, argv + argc);
-    as.emplace_back("--float");
+    // as.emplace_back("--float");
```
Comment (atamazov) [inline test/driver.hpp:1389]: Why this is necessary?

File: test/immed_conv2d.cpp
Comment (atamazov) [inline test/immed_conv2d.cpp:1]: This must be reverted. See https://github.com/ROCm/MIOpen/pull/2545#discussion_r1423209428

File: test/pooling2d.hpp
Comment (atamazov) [inline test/pooling2d.hpp:1]: This must be reverted. See https://github.com/ROCm/MIOpen/pull/2545#discussion_r1423209428

File: test/conv3d.hpp
Comment (atamazov) [inline test/conv3d.hpp:1]: This must be reverted too due to the same reason.

File: test/gtest/pooling2d_asymmetric.cpp [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+#include "pooling2d.hpp"
+#include "get_handle.hpp"
+#include <miopen/env.hpp>
+#include <gtest/gtest.h>
+
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_ALL)
+MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLAGS_ARGS)
+
+static bool SkipTest(void) { return miopen::IsDisabled(ENV(MIOPEN_TEST_ALL)); }
```
Comment (atamazov) [inline test/gtest/pooling2d_asymmetric.cpp:35]: Contributes to the issue #1843. Don't be surprised if CI performance is dropping.

File: test/gtest/pooling2d_wide.cpp [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+#include "pooling2d.hpp"
+#include "get_handle.hpp"
+#include <miopen/env.hpp>
+#include <gtest/gtest.h>
+
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_ALL)
+MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLAGS_ARGS)
+
+static bool SkipTest(void) { return miopen::IsDisabled(ENV(MIOPEN_TEST_ALL)); }
```
Comment (atamazov) [inline test/gtest/pooling2d_wide.cpp:35]: Contributes to the issue https://github.com/ROCm/MIOpen/issues/1843.

**Review by atamazov — COMMENTED** [review]
🔴 Silently removes 
- `test_conv3d -all`
- `test_immed_conv2d -all`
- `test_pooling2d -all`

**Comment by atamazov** [comment]
@junliume @JehandadKhan This must be reverted ASAP.

/cc @shurale-nkn

**Comment by junliume** [comment]
This PR is reverted as per #2616 and should start some rework. Thanks!

**Pull request merged**